### ISSUE TITLE
ID 261621: [Keyboard Navigation - Simon – Alert rules]: Keyboard Focus does not return to “Search by data source” combo box after pressing esc key.

### ISSUE THREAD

**Issue by eledobleefe** [grafana/grafana#72826]
### **User Experience**:  
Users who rely on the keyboard will be affected if the keyboard focus is lost after invoking the ‘ESC' key on. Due to this issue, the user will be misguided and unable to track the keyboard focus.
_Note: User credentials should NOT be included in the bug._

### **Repro Steps**:
1. Open #Url: [play.grafana.org](https://play.grafana.org/) Sign in with valid credentials. "Home" page appears. 
2. Navigate to the left navigation bar and to the "Alerting" button, select it. It gets expanded and navigates to the expanded sub menu items. 
3. Select the "Alert Rules" menu items. A new page "Alerting" appears. 
4. Navigate the tab " Alert Rules". 
5. Navigate to "Search by data source” combo box. 
6. Observe and verify the issue

**Actual Result**: 
Issue: Keyboard Focus does not return to “Search by data source” combo box after pressing esc key.
Observation: When focus is inside the "Search by data source" combo box dialog,after invoking 'ESC' key keyboard focus disappears. Further pressing tab key, the keyboard focus appears on next control "Search by label" text box.

**Expected Result**:

**Solution:** 
Keyboard Focus should return to “Search by data source” combo box after pressing esc key. EX: When focus is inside the "Search by data source" combo box dialog, after invoking 'ESC' key keyboard focus should return to "Search by data source" combo box.

### **Severity:** 
`SEV3`

### **Area:**
`MultipleDataSourcePicker Component (Alert Rules)`: alerting-experience
_Note: The same issue exists with 'Click to view alternative operations' combo box under explore screen. Same issue exists under 'Save' dashboard dialog box._
`QueryEditor`: grafana-explore

**Comment by Elfo404** [comment]
removing this from the explore project as mentioned in the description the issue happens in alerting and not in Explore. @grobinson-grafana any reason you removed it from the Alerting project?

**Comment by samsch** [comment]
It looks like this is not longer an issue! Unsure when it was fixed, but looks good now
